x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/expression/predicate/operator/arithmetic/Sub.java

```diff
@@ -44,7 +44,7 @@ public Sub(Source source, Expression left, Expression right) {
     @Override
     protected TypeResolution resolveType() {
         TypeResolution resolution = super.resolveType();
-        if (resolution.resolved() && EsqlDataTypes.isDateTimeOrTemporal(dataType()) && DataTypes.isDateTime(left().dataType()) == false) {
+        if (resolution.resolved() && EsqlDataTypes.isDateTimeOrTemporal(dataType()) && DataTypes.isDateTime(right().dataType())) {
```

Comment:
The changed format works because `super.resolveType()` ensures that they the operands are of different type, but the original form makes it more clear about the LHS type requirement.

Reply:
Hm, to allow folding of durations/periods with SUB, I need to allow the left hand side to be something different than datetime. In this respect, the two conditions are not equivalent and I need this change to allow e.g. `1week - 1day` while still disallowing `1week - now()`. This is the most concise way I came up with, and it seems logical to me: it says "you cannot subtract a datetime from something else".

I'll add a comment to make sure this is as clear as can be; if there's something else I can do to make things clear, let me know!